gitter.py

```diff
@@ -179,7 +180,14 @@ def idd(self):
     def name(self):
         return self._name
 
-    joined = True  # TODO
+    @property
+    def joined(self):
+        return self._joined
+
+    @joined.setter
+    def joined(self, value):
+        self._joined = value
```

Comment:
I don't think this setter should be here. We only want outside code to get the property, not set it. It's only ever set internally from within this class.